it/topics/scripting_languages/JavaScript/frameworks.md

```diff
@@ -0,0 +1,393 @@
+# JS Frameworks
+
+Le strutture Javascript sono librerie (solitamente disponibili come pacchetti _npm_) che ti consentono di creare frontend animati, dinamici, interattivi e moderni utilizzando astrazioni come i componenti in modo da non dover scrivere JavaScript di basso livello.
+
+Vedi [10 migliori framework JavaScript da utilizzare nel 2019](https://hackr.io/blog/10-best-javascript-frameworks-2019).
+
+Di seguito sono riportati i riepiloghi di alcuni framework JavaScript popolari, inclusi collegamenti e frammenti di codice _hello world_. Il focus di questo elenco è sui framework per la creazione di app Web, sia sul frontend(rendering lato client) sia sul backend(rendering server-sider).
+
+
+## React
+
+### Perché React?
+
+Il motivo principale è che React ha un linguaggio dichiarativo per modellare l'interfaccia utente e lo stato.
+
+Lo scopo di React:
+
+- Descrivere in modo dichiarativo le interfacce utente (UI) e modellare lo stato di queste interfacce.
+- vale a dire, usalo descrivendo come dovrebbe essere un'interfaccia o uno stato finito e React capirà come arrivare a quello.
+- Ciò consente agli sviluppatori di risparmiare molto tempo e poiché non è necessario controllare lo stato prima di apportare modifiche a livello di transazione.
+
+Vedi [Perchè React?](https://jscomplete.com/learn/why-react) sul sito JSComplete.
+
+### Informazioni su React
+
+- Viene utilizzato per sviluppare un'interfaccia utente dinamica in modo che il DOM sia basato sul _client side_ - come applicazione a pagina singola (SPA).
+- Questo si adatta bene per il traffico ad alto volume.
+- Funziona bene con dati dinamici come la lettura o l'archiviazione da un back-end o database o l'interrogazione di altri servizi API.
+- È molto popolare (la quinta libreria JS più famosa su Github) e utilizzata sui principali siti Facebook, Netflix e Khan Academy.
+- React non si definisce un "framework" ma solo una "libreria". È facile da estendere, non supponente ed è modulare (puoi usare solo una parte di cui hai bisogno senza tutto).
+- React è una libreria per la creazione di _user interfaces_. Se un ambiente come il browser Web può eseguire JavaScript semplice, è possibile creare un'interfaccia per esso utilizzando React.
+- React verrà compilato in codice JavaScript di basso livello ottimizzato.
+- React è "solo JavaScript" - sì, devi imparare l'API React ma puoi usare JavaScript semplice all'interno di React e il risultato è JavaScript. vale a dire dopo aver compilato in una directory di build, è possibile servire l'applicazione _senza_Reagire installato. E non devi caricare React sul lato browser, ma solo il file JavaScript in bundle (webpack aiuta a questo scopo).
```

Comment:
Check the underscores and spacing on `_senza_Reagire`.  The italics needs to be resolved.